bundles/org.openhab.binding.openthermgateway/README.md

```diff
@@ -1,195 +1,432 @@
+        Switch item="ServiceRequest" icon="switch" label="Service Request"
+        Switch item="ExhaustFanFault" icon="switch" label="Exhaust Fan Fault"
+        Switch item="InletFanFault" icon="switch" label="Inlet Fan Fault"
+        Switch item="FrostProtection" icon="switch" label="Frost Protection"
+        Text item="FaultCode" label="Fault Code"
+        Text item="DiagnosticCode" label="Diagnostic Code"
+        Text item="SystemType" label="System Type"
+        Switch item="Bypass" icon="switch" label="Bypass"
+        Text item="SpeedControl" label="Speed Control"
+        Text item="MemberID" label="Member ID"
+        Text item="OpenThermVersion" label="OpenTherm Version"
+        Text item="VersionType" label="Version Type"
+        Text item="RelativeVentilation" label="Relative Ventilation [%d %%]"
+        Text item="RelativeHumidity" label="Relative Humidity [%d %%]"
+        Text item="CO2Level" label="CO2 Level [%d ppm]"
+        Text item="SupplyInletTemperature" label="Supply Inlet Temperature [%.1f %unit%]"
+        Text item="SupplyOutletTemperature" label="Supply Outlet Temperature [%.1f %unit%]"
+        Text item="ExhaustInletTemperature" label="Exhaust Inlet Temperature [%.1f %unit%]"
+        Text item="ExhaustOutletTemperature" label="Exhaust Outlet Temperature [%.1f %unit%]"
+        Text item="ActualExhaustFanSpeed" label="Actual Exhaust Fan Speed [%d rpm]"
+        Text item="ActualInletFanSpeed" label="Actual Inlet Fan Speed [%d rpm]"
+        Switch item="NominalVentilationValueTransfer" icon="switch" label="Nominal Ventilation Value Transfer"
+        Text item="NominalVentilationValue" label="Nominal Ventilation Value"
+        Text item="NominalVentilationValue" label="Nominal Ventilation Value [%d %%]"
+        Text item="VentilationSetpoint" label="Ventilation Setpoint" minValue="0" maxValue="100" step="1"
+        Text item="TransparentSlaveParameterNumber" label="Transparent Slave Parameter Number"
+        Text item="TransparentSlaveParameterEntry" label="Transparent Slave Parameter Entry"
+        Text item="FaultHistoryBufferNumber" label="Fault History Buffer Number"
+        Text item="FaultHistoryBufferEntry" label="Fault History Buffer Entry"
+    }
+}
+```
+
+## Transparent Slave Parameters and Fault History Buffer channels
+
+The transparent slave parameters (TSP) and fault history buffer (FHB) use a variable number of entries.
+The number of entries is determined by a TSP or FHB size message.
+Channels for TSP and FHB entries are automatically created when a TSP or FHB size message is received.
+An index number is added to the base channel name to create a unique channel name for each entry.
+
+For example, if a TSP size message is received for a boiler unit (OpenTherm DATA-ID 10) with value 60, then channels `tspentry_0` through `tspentry_59` will be automatically created and linked to the corresponding TSP entry (OpenTherm DATA-ID 11).
```

Comment:
How are these Channels used? Do you need a sufficient number of Items configured to display the values in the UI?

Reply:
The number of TSP and FHB channels is different for each device. The gateway first requests the number of TSP and FHB values and then proceeds to fetch each value. You would indeed need corresponding Items if you would like to display the values in the UI. Based on documentation of the device, a user should be able to tell how many values there are and what each value means. They may choose to create Items only for those channels that are of interest.

Reply:
Ok, if the Channel count is constant it should be fine.